Microsoft.Toolkit.Uwp.SampleApp/SamplePages/RichSuggestBox/RichSuggestBoxPage.xaml.cs

```diff
@@ -0,0 +1,192 @@
+            }
+
+            if (control.FindChild("SuggestingBox") is RichSuggestBox rsb)
+            {
+                this._rsb = rsb;
+                this._rsb.SuggestionChosen += this.SuggestingBox_OnSuggestionChosen;
+                this._rsb.SuggestionsRequested += this.SuggestingBox_OnSuggestionsRequested;
+            }
+
+            if (control.FindChild("PlainTextSuggestingBox") is RichSuggestBox tsb)
+            {
+                this._tsb = tsb;
+                this._tsb.SuggestionChosen += this.SuggestingBox_OnSuggestionChosen;
+                this._tsb.SuggestionsRequested += this.SuggestingBox_OnSuggestionsRequested;
+                this._tsb.TokenHovering += this.SuggestingBox_OnTokenHovering;
+            }
+
+            if (control.FindChild("TokenListView1") is ListView tls1)
+            {
+                tls1.ItemsSource = this._rsb?.Tokens;
+            }
+
+            if (control.FindChild("TokenListView2") is ListView tls2)
+            {
+                tls2.ItemsSource = this._tsb?.Tokens;
+            }
+        }
+
+        private void SuggestingBox_OnTokenHovering(RichSuggestBox sender, RichSuggestTokenHoveringEventArgs args)
+        {
+            var flyout = (Flyout)FlyoutBase.GetAttachedFlyout(sender);
+            var pointerPosition = args.CurrentPoint.Position;
+
+            if (flyout?.Content is ContentPresenter cp && sender.TextDocument.Selection.Type != SelectionType.Normal &&
+                (!flyout.IsOpen || cp.Content != args.Token.Item))
+            {
+                this._dispatcherQueue.TryEnqueue(() =>
+                {
+                    cp.Content = args.Token.Item;
+                    flyout.ShowAt(sender, new FlyoutShowOptions
+                        {
+                            Position = pointerPosition,
+                            ExclusionRect = sender.GetRectFromRange(args.Range),
+                            ShowMode = FlyoutShowMode.TransientWithDismissOnPointerMoveAway,
+                        });
+                });
+            }
+        }
+
+        private void SuggestingBox_OnSuggestionChosen(RichSuggestBox sender, SuggestionChosenEventArgs args)
+        {
+            if (args.Prefix == "#")
+            {
+                args.Format.BackgroundColor = Colors.DarkOrange;
+                args.Format.ForegroundColor = Colors.OrangeRed;
+                args.Format.Bold = FormatEffect.On;
+                args.Format.Italic = FormatEffect.On;
+                args.Text = ((SampleDataType)args.SelectedItem).Text;
+            }
+            else
+            {
+                args.Text = ((SampleEmailDataType)args.SelectedItem).DisplayName;
+            }
+        }
+
+        private void SuggestingBox_OnSuggestionsRequested(RichSuggestBox sender, SuggestionsRequestedEventArgs args)
+        {
+            if (args.Prefix == "#")
+            {
+                sender.ItemsSource =
```

Comment:
Ah, I missed how this was setup before. If you look at the TokenizingTextBox and its sample. We use the underlying `ItemsSource` from `ListViewBase` (or in your case `ItemsControl`) to represent the tokens present in the textbox. Here that's your `Tokens` property. We then have a separate `SuggestedItemsSource` for the source of the drop-down when a search is made. This allows the developer to data-bind to the ItemsSource and set things programmatically (and where we utilize the `AdvancedCollectionView`):

https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/5171b99c95323a704922a6bdc70a1a5d5eb650cf/Microsoft.Toolkit.Uwp.SampleApp/SamplePages/TokenizingTextBox/TokenizingTextBoxPage.xaml.cs#L133

Of course with RichSuggestBox this is a bit different as there's a bunch of other things going on beyond the underlying `ItemsSource` and a developer may just serialize the string/rtf data and back. Have you thought about how you could programmatically re-populate the control?

I could also think of a scenario like Outlook where a user adds an attachment in the e-mail, but there's still a list of attachments elsewhere and the user could remove it from that list.

_I'm not entirely sure what makes sense here_... As unlike the TokenizingTextBox, the 'Item' level here is really still all about the suggestions for the ItemTemplate/Selector as well. Would it make sense to swap the usage here and have the `ItemsSource` represent the selected tokens within the document and then use a `SuggestedItemsSource` to manipulate for the drop-down? Or just keep it as is and maybe elaborate what `Tokens` are? @shweaver-MSFT I know you thought about TokenizingTextBox a bit in the past too, what are your thoughts here?

Whether or not the document reacts to changes to the underlying `ItemsSources` if the dev deletes things is probably not as important right now; however, in either case you can just have the ListView showing the token bind directly to the RichSuggestBox.ItemsSource rather than setting them in code-behind (farther up above):

https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/5171b99c95323a704922a6bdc70a1a5d5eb650cf/Microsoft.Toolkit.Uwp.SampleApp/SamplePages/TokenizingTextBox/TokenizingTextBoxXaml.bind#L68

Reply:
@michael-hawker The `Tokens` property is separated from the `ItemsSource` is by design. The `Tokens` property must have the type of `ReadOnlyObservableCollection`. This is important to prevent dev from manually deleting any tokens. Doing so would make the document go out of sync with the token list, which may lead to a confusing end-user experience, especially when the control is configured in plain-text only mode (second example in the sample page). 

Another reason is the `Tokens` collection is always a collection of `RichSuggestToken`, I think it makes sense to have its own property rather than relying on `ItemsSource` which has a generic `object` type,

I have not thought of programmatically repopulating the control before, but one way I can think of from the top of my head is to add a `Load()` method that takes in RTF data and a collection of `RichSuggestToken`s. That way we can make sure the document and the internal token list are in sync.

Reply:
I have reworked the flow of this suggestion request a little bit to support something like `AdvancedCollectionView`. So you don't need to keep changing `ItemsSource` to update the suggestion list anymore. The new flow is: after the `SuggestionsRequested` event is triggered, the suggestion popup **_always_** shows up unless the dev explicitly cancels it (through the event args) or if the suggestion list is empty.